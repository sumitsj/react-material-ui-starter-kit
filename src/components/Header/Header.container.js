import React from 'react';
import { useStateValue } from '../../store';
import Header from './Header.view';

const HeaderContainer = () => {
  const [{ name }] = useStateValue();
  return <Header name={name} />;
};

export default HeaderContainer;
