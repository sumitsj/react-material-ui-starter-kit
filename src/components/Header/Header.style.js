import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() => ({
  logoLink: {
    color: '#fff',
  },
}));

export default useStyles;
