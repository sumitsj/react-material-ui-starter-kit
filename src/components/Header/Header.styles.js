import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() => ({
  toolbar: {
    justifyContent: 'space-between',
  },
  logoLink: {
    color: '#fff',
  },
}));

export default useStyles;
