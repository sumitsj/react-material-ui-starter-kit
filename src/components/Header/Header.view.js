import React from 'react';
import AppBar from '@material-ui/core/AppBar/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Link from '@material-ui/core/Link';
import * as PropTypes from 'prop-types';
import useStyles from './Header.style';

const Header = ({ name }) => {
  const classes = useStyles();

  return (
    <AppBar>
      <Toolbar>
        <Link href="/" variant="body1" className={classes.logoLink}>
          {name}
        </Link>
      </Toolbar>
    </AppBar>
  );
};

Header.propTypes = {
  name: PropTypes.string.isRequired,
};

export default Header;
