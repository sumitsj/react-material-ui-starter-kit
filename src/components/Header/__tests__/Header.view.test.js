import React from 'react';
import { render } from '@testing-library/react';
import Header from '../Header.view';

jest.mock('../../HeaderMenu', () => jest.fn(() => <div title="header menu" />));

describe('Header', () => {
  describe('HTML Structure', () => {
    it('should render properly', () => {
      const { asFragment } = render(<Header name="Demo" isAuthenticated={false} />);
      expect(asFragment()).toMatchSnapshot();
    });

    it('should render with user menu if authenticated', () => {
      const { asFragment } = render(<Header name="Demo" isAuthenticated />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
