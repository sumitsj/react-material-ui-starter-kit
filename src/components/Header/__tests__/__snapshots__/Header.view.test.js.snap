// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header HTML Structure should render properly 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineNone makeStyles-logoLink-1 MuiTypography-body1 MuiTypography-colorPrimary"
        href="/"
      >
        Demo
      </a>
    </div>
  </header>
</DocumentFragment>
`;
